### ISSUE TITLE
Inconsistency between QtPluginTable and list_plugin_metadata

### ISSUE THREAD

**Issue by Czaki** [napari/napari#1792]
## 🐛 Bug

When napari try to get metadata of installed plugins it ignores values which cannot be read:

https://github.com/napari/napari-plugin-engine/blob/bf7ec61a06eb7d2765c7cd68915187e2c4a4222f/napari_plugin_engine/manager.py#L801-L806

```python
        for caller in self._plugin2hookcallers[plugin]:
            try:
                impl = caller.get_plugin_implementation(plugin_name)
                hooks.append(impl.specname)
            except KeyError:
                pass
```

But when adding data to the table in QtPluginTable it requires all headers visible:

https://github.com/napari/napari/blob/b576203c58fc71d06460811a6a0eff8fb7191061/napari/_qt/widgets/qt_dict_table.py#L85-L92

and 

https://github.com/napari/napari/blob/b576203c58fc71d06460811a6a0eff8fb7191061/napari/_qt/dialogs/qt_plugin_table.py#L25-L37

This creates a problem when some of the metadata cannot be read. 
In my case, my application PartSeg is frozen using Pyinstaller. Pyinstaller does not keep package metadata so i need to manually call register method of plugin manager:

https://github.com/4DNucleome/PartSeg/blob/dc5ce1aeac80018f1e78dc49d7fa559090d52ce9/package/PartSeg/plugins/__init__.py#L29

But this way plugin manager cannot get all data and trying of see plugin list ends with 

```
ValueError: Argument 'headers' got item 'package', which was not found in any of the items in 'data'
  File "site-packages/napari/_qt/qt_main_window.py", line 294, in _show_plugin_list
  File "site-packages/napari/_qt/dialogs/qt_plugin_table.py", line 36, in __init__
  File "site-packages/napari/_qt/widgets/qt_dict_table.py", line 60, in __init__
  File "site-packages/napari/_qt/widgets/qt_dict_table.py", line 89, in set_data
```

where variable values are:

```
_headers | ['hooks', 'plugin_name']
-- | --
data | [{hooks: ['napari_get_writer', 'napari_write_image', 'napari_write_labels', 'napari_write_points', 'napari_write_shapes', 'napari_write_vectors'], plugin_name: 'napari_svg'}, {hooks: ['napari_get_reader'], plugin_name: 'PartSegCore.napari_plugins.load_image'}, {hooks: ['napari_get_reader'], plugin_name: 'PartSegCore.napari_plugins.load_mask_project'}, {hooks: ['napari_get_reader'], plugin_name: 'PartSegCore.napari_plugins.load_masked_image'}, {hooks: ['napari_get_reader'], plugin_name: 'PartSegCore.napari_plugins.load_roi_project'}, {hooks: [], plugin_name: 'napari_plugin_engine'}]
h | 'package'
headers | ['plugin_name', 'package', 'version', 'url', [Filtered], 'license']
nrows | 6
```

## To Reproduce

1. Download bundled application https://github.com/4DNucleome/PartSeg/releases/tag/v0.12.0
2. Open: View > Napari viewer 
3. Open: Plugins > List installed plugins 

## Additional context

Current release of Breifcase does not allow to bundle my application. My PR fixing problem with the upper case letter in name of application was merged, but there is no new release after this.

**Comment by tlambert03** [comment]
QtPluginTable would be removed/superceded by #1357

**Comment by sofroniewn** [comment]
This has been closed by #1357
